src/Controller/IndexController.php

```diff
@@ -87,75 +86,78 @@ public function indexAction(LocalBusinessRepository $repository, CacheInterface
             $cacheKeySuffix = 'anonymous';
         }
 
-        $countByType = $repository->countByType();
-        $types = array_keys($countByType);
-
-        $typesWithEnoughShops = [];
-        foreach ($countByType as $type => $countForType) {
-            if ($countForType >= self::ITEMS_PER_ROW || $countForType >= (self::ITEMS_PER_ROW * 2)) {
-                $typesWithEnoughShops[] = $type;
-            }
-        }
-
         $inflector = new EnglishInflector();
 
         $sections = [];
 
-        if (count($typesWithEnoughShops) > 0) {
-
-            foreach ($typesWithEnoughShops as $type) {
-
-                $keyForType = LocalBusiness::getKeyForType($type);
+        $restaurantType = FoodEstablishment::RESTAURANT;
+        $keyForRestaurantType = LocalBusiness::getKeyForType($restaurantType);
+        $cacheKey = sprintf('homepage.%s.%s', $keyForRestaurantType, $cacheKeySuffix);
+
+        [ $restaurants, $restaurantsCount ] =
+            $this->getItems($repository, $restaurantType, $projectCache, $cacheKey, $timingRegistry);
+
+        if ($restaurantsCount > 0) {
+            $sections[] = [
+                'type' => $restaurantType,
+                'shops' => $restaurants,
```

Comment:
Here we are passing all the restaurants to the view, which may impact performance, and makes the "view all" button useless. It should be limited somehow.